library/core/src/cell.rs

```diff
@@ -793,7 +803,17 @@ impl<T: ?Sized> RefCell<T> {
     #[inline]
     #[track_caller]
     pub fn borrow(&self) -> Ref<'_, T> {
-        self.try_borrow().expect("already mutably borrowed")
+        // FIXME: Use `unwrap_or_else` once we support `#[track_caller]` on closures
+        #[cfg(feature = "debug_refcell")]
+        match self.try_borrow() {
+            Ok(b) => return b,
+            Err(e) => {
+                panic!("already mutably borrowed: {:?} at {}", e, self.first_caller.get().unwrap())
+            }
+        }
+
+        #[cfg(not(feature = "debug_refcell"))]
+        return self.try_borrow().expect("already mutably borrowed");
```

Comment:
Wouldn't it be better if the `cfg` only covered the actual panic line? As it stands there's "hidden" duplication across feature gated state.